src/main/java/org/openRealmOfStars/player/tech/TechList.java

```diff
@@ -902,50 +825,61 @@ public double getTechResearchPoints(final TechType type) {
   public void setTechResearchPoints(final TechType type, final double value) {
     techResearchPoint[type.getIndex()] = value;
   }
+
   /**
    * Add new random tech to player tech list
    * @param info PlayerInfo
    * @return Added Tech or null if all tech has been invented
    */
   public Tech addNewRandomTech(final PlayerInfo info) {
     Tech result = null;
-    int loop = 10;
-    while (result == null && loop > 0) {
-      loop--;
+    for (int j = 0; result == null && j < 10; j += 1) {
       int index = DiceGenerator.getRandom(0, 5);
       TechType type = TechType.getTypeByIndex(index);
       int lvl = techLevels[index];
       Tech tech = TechFactory.createRandomTech(type, lvl,
           getListForTypeAndLevel(type, lvl), race);
+
       if (tech == null) {
         if (lvl < 10) {
```

Comment:
This could be MAX_TECH_LEVELS. It wasn't in original but this would improve could readability.